src/core/CONVENTIONS.md

```diff
@@ -347,10 +347,10 @@ export const myType: SavedObjectsType = {
       },
     },
   },
-  migrations: {
+  migrations: () => ({
```

Comment:
Unless we want to deprecate the `SavedObjectMigrationMap` return and only accept a function I think our docs should continue to show the static migration map object since 99% of plugins will be fine using the static object.